### PR TITLE
Uses Magma 1.8

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

- Uses container images from Magma 1.8
- Adds ports that are required in 1.8
- Adds new relations
- Adds storage where required
- Uses new main organisation name in nms-magmalte
- Adds rootca to bootstrapper
- Modifies startup commands

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

File: orc8r-accessd-operator/metadata.yaml [hunk 1]
```diff
@@ -12,12 +12,27 @@ website: https://github.com/canonical/charmed-magma-orchestrator
...
+
+storage:
+  config:
+    type: filesystem
+    description: Configs storage
+    minimum-size: 1M
+  certs:
+    type: filesystem
+    description: Certs storage
+    minimum-size: 1M
```
Comment (Gmerold) [inline orc8r-accessd-operator/metadata.yaml]: Do we need it here? I don't see any new charm code which would make use of it.

File: orc8r-bootstrapper-operator/metadata.yaml [hunk 2]
```diff
@@ -16,14 +16,20 @@ containers:
     mounts:
       - storage: certs
         location: /var/opt/magma/certs
+      - storage: config
+        location: /var/opt/magma/configs/orc8r
```
Comment (Gmerold) [inline orc8r-bootstrapper-operator/metadata.yaml]: Do we need this?

File: orc8r-bootstrapper-operator/metadata.yaml [hunk 3]
```diff
@@ -16,14 +16,20 @@ containers:
...
     type: oci-image
     description: OCI image for magma-orc8r-bootstrapper
-    upstream-source: ghcr.io/canonical/magma-orc8r-controller:1.6.1
+    upstream-source: ghcr.io/canonical/magma-orc8r-controller:1.8.0
 
 storage:
+  config:
+    type: filesystem
+    description: Configs storage
+    minimum-size: 1M
```
Comment (Gmerold) [inline orc8r-bootstrapper-operator/metadata.yaml]: As above.

File: orc8r-bootstrapper-operator/README.md [hunk 4]
```diff
@@ -29,4 +29,4 @@ juju relate orc8r-bootstrapper orc8r-certifier
 
 ## OCI Images
 
-Default: linuxfoundation.jfrog.io/magma-docker/controller:1.6.0
+Default: ghcr.io/canonical/magma-orc8r-controller:1.8.0
```
Comment (Gmerold) [inline orc8r-bootstrapper-operator/README.md:36]: Not related to this line, but the whole README could use some update :)
It should reflect the fact that the bootstrapper needs the DB now, also we should specify the relation name for `juju relate`. `Requires` section should be updated too.

File: orc8r-bundle/tests/unit/expected_bundles/local.yaml [hunk 5]
```diff
@@ -48,103 +48,103 @@ applications:
...
     resources:
-      magma-orc8r-analytics-image: linuxfoundation.jfrog.io/magma-docker/controller:1.6.0
+      magma-orc8r-analytics-image: ghcr.io/canonical/magma-orc8r-controller:1.8.0
     scale: 1
     trust: true
   orc8r-bootstrapper:
     charm: ./magma-orc8r-bootstrapper_ubuntu-20.04-amd64.charm
     resources:
-      magma-orc8r-bootstrapper-image: linuxfoundation.jfrog.io/magma-docker/controller:1.6.0
+      magma-orc8r-bootstrapper-image: ghcr.io/canonical/magma-orc8r-controller:1.8.0
```
Comment (Gmerold) [inline orc8r-bundle/tests/unit/expected_bundles/local.yaml:57]: What about new relations required by the bootstrapper?

**Review by Gmerold — CHANGES_REQUESTED** [review]

File: orc8r-accessd-operator/metadata.yaml (hunk 1 above)
Comment (saltiyazan) [reply orc8r-accessd-operator/metadata.yaml]: Removed.

File: orc8r-bootstrapper-operator/metadata.yaml (hunk 2 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/metadata.yaml]: Removed.

File: orc8r-bootstrapper-operator/metadata.yaml (hunk 3 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/metadata.yaml]: Removed.

File: orc8r-bundle/tests/unit/expected_bundles/local.yaml (hunk 5 above)
Comment (saltiyazan) [reply orc8r-bundle/tests/unit/expected_bundles/local.yaml:57]: Done.

File: orc8r-bootstrapper-operator/README.md (hunk 4 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/README.md:36]: ✅

File: orc8r-bootstrapper-operator/src/charm.py [hunk 6]
```diff
@@ -5,44 +5,74 @@
...
-from ops.model import ActiveStatus, ModelError, Relation, WaitingStatus
+from ops.model import ActiveStatus, BlockedStatus, ModelError, Relation, WaitingStatus
 from ops.pebble import Layer
+from pgconnstr import ConnectionString  # type: ignore[import]
 
 from private_key import generate_private_key
 
 logger = logging.getLogger(__name__)
 
+pgsql = ops.lib.use("pgsql", 1, "postgresql-charmers@lists.launchpad.net")
```
Comment (ghislainbourgeois) [inline orc8r-bootstrapper-operator/src/charm.py:36]: This is a deprecated call, is it possible to import the lib directly?

File: orc8r-bootstrapper-operator/tests/integration/test_integration.py [hunk 7]
```diff
@@ -3,22 +3,86 @@
...
+        await self._deploy_tls_certificates_operator(ops_test)
+        await self._deploy_orc8r_certifier(ops_test)
+
+    @staticmethod
+    def _find_charm(charm_dir: str, charm_file_name: str) -> Optional[str]:
+        for root, _, files in os.walk(charm_dir):
+            for file in files:
+                if file == charm_file_name:
+                    return os.path.join(root, file)
+        return None
```
Comment (ghislainbourgeois) [inline orc8r-bootstrapper-operator/tests/integration/test_integration.py]: `Path.rglob()` would be a cleaner solution for this: https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: orc8r-bootstrapper-operator/README.md [hunk 8]
```diff
@@ -13,6 +13,8 @@ be deployed as a bundle.
 ```bash
 juju deploy magma-orc8r-bootstrapper orc8r-bootstrapper
 juju relate orc8r-bootstrapper orc8r-certifier
```
Comment (Gmerold) [inline orc8r-bootstrapper-operator/README.md]: This relation was actually never needed and can be removed.

File: orc8r-bootstrapper-operator/README.md [hunk 9]
```diff
@@ -13,6 +13,8 @@ be deployed as a bundle.
...
+juju relate orc8r-bootstrapper postgresql-k8s:db
```
Comment (Gmerold) [inline orc8r-bootstrapper-operator/README.md:17]: It used to be a convention that if a charm requires other charms to work, we would mention deploy and relate commands for all of them; just to provide a ready solution for a working setup. I see it's been kinda neglected, but I think it's worth doing that. WDYT?

File: orc8r-bootstrapper-operator/README.md [hunk 10]
```diff
@@ -22,11 +24,13 @@ juju relate orc8r-bootstrapper orc8r-certifier
 ### Requires
 
 - **cert-bootstrapper**: Relation that provides the bootstrapper private key.
```
Comment (Gmerold) [inline orc8r-bootstrapper-operator/README.md]: Private key is generated by the charm internally. There's nothing that provides it.

File: orc8r-bootstrapper-operator/src/charm.py (hunk 6 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/src/charm.py:36]: This [lib here](https://github.com/canonical/postgresql-k8s-operator/tree/main/lib/charms/postgresql_k8s/v0) that can be imported directly has a different interface :)
I know we plan to use this library. But for now to use the same like in other charms I went with this call.
Let me know if you don't agree :)

File: orc8r-bootstrapper-operator/tests/integration/test_integration.py (hunk 7 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/tests/integration/test_integration.py]: ✅

File: orc8r-bootstrapper-operator/README.md (hunk 8 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/README.md]: Done

File: orc8r-bootstrapper-operator/README.md (hunk 10 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/README.md]: Done.

File: orc8r-bootstrapper-operator/README.md (hunk 9 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/README.md:17]: Done.
I modified the README in orc8r-nginx too, to follow that.

**Review by Gmerold — APPROVED** [review]

File: orc8r-bootstrapper-operator/src/charm.py (hunk 6 above)
Comment (ghislainbourgeois) [reply orc8r-bootstrapper-operator/src/charm.py:36]: OK, that is fine

File: orc8r-bootstrapper-operator/tests/integration/test_integration.py [hunk 11]
```diff
@@ -4,21 +4,82 @@
...
+    async def setup(self, ops_test):
+        await self._deploy_postgresql(ops_test)
+        await self._deploy_tls_certificates_operator(ops_test)
+        await self._deploy_orc8r_certifier(ops_test)
+
+    @staticmethod
+    def _find_charm(charm_dir: str, charm_file_name: str) -> Optional[str]:
+        for path in Path(charm_dir).rglob(charm_file_name):
+            return str(path)
+        return None
```
Comment (ghislainbourgeois) [inline orc8r-bootstrapper-operator/tests/integration/test_integration.py]:
```suggestion
        if path := Path(charm_dir).rglob(charm_file_name):
            return str(path[0])
        return None
```

Small nitpick

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: orc8r-bootstrapper-operator/tests/integration/test_integration.py (hunk 11 above)
Comment (saltiyazan) [reply orc8r-bootstrapper-operator/tests/integration/test_integration.py]: the function returns a generator object that we can't index. and if I use `next` directly on it, there is a chance of gettin `StopIteration` exception.
I solved as follows:
```
def _find_charm(charm_dir: str, charm_file_name: str) -> Optional[str]:
        try:
            path = next(Path(charm_dir).rglob(charm_file_name))
            return str(path)
        except StopIteration:
            return None
```
 Please take a look and let me know.

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**